src/config.rs

```diff
@@ -675,17 +675,13 @@ impl<'a> Cfg<'a> {
 
                     // XXX: this awkwardness deals with settings file being locked already
                     let toolchain_name = toolchain_name.resolve(&default_host_triple)?;
-                    match Toolchain::new(self, (&toolchain_name).into()) {
```

Comment:
I have manually inlined `Toolchain::new()` here and removed the statement causing an infinite recursion, because to allow any error to happen with this call, the toolchain must not exist:

https://github.com/rust-lang/rustup/blob/3b60eb245d06d9930c9524640130394759f42808/src/toolchain.rs#L107-L119